src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,443 @@
+                str(enable_hugepage) +
+                '|" ' +
+                file_path)
+            cmds.append(
+                'sed -i "s|dellnfv::numa::enable:.*|dellnfv::numa::enable: ' +
+                str(enable_numa) +
+                '|" ' +
+                file_path)
+
+            if dell_compute_count > 0:
+                cmds.append(
+                    'sed -i "s|DellComputeCount:.*|DellComputeCount: ' +
+                    str(dell_compute_count) +
+                    '|" ' +
+                    file_path)
+
+            if enable_hugepage:
+                hugepage_number = ConfigOvercloud.calculate_hugepage_count(
+                    hugepage_size)
+                cmds.append('sed -i "s|dellnfv::hugepages::hugepagesize:.*'
+                            '|dellnfv::hugepages::hugepagesize: ' +
+                            hugepage_size[0:-1] + '|" ' + file_path)
+                cmds.append(
+                    'sed -i "s|dellnfv::hugepages::hugepagecount:.*'
+                    '|dellnfv::hugepages::hugepagecount: ' +
+                    str(hugepage_number) +
+                    '|" ' +
+                    file_path)
+            if enable_numa:
+                vcpu_pin_set = ConfigOvercloud.calculate_hostos_cpus(
+                    hostos_cpu_count)
+                cmds.append(
+                    "sed -i 's|dellnfv::numa::vcpu_pin_set:.*"
+                    "|dellnfv::numa::vcpu_pin_set: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+                cmds.append(
+                    'sed -i "s|  # NovaVcpuPinSet|  ' +
+                    'NovaVcpuPinSet|" ' +
+                    file_path)
+                cmds.append(
+                    "sed -i 's|NovaVcpuPinSet:.*|NovaVcpuPinSet: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+
+            for cmd in cmds:
+                status = os.system(cmd)
+                logger.debug("cmd: {}".format(cmd))
+            if status != 0:
+                raise Exception(
+                    "Failed to execute the command {}"
+                    " with error code {}".format(
+                        cmd, status))
+
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to modify the dell_environment.yaml"
+                " at location {}".format(file_path))
+
+    def get_dell_compute_nodes_hostnames(self, nova):
+        try:
+            logger.info("Getting dellnfv compute node hostnames")
+
+            # Create host object
+            host_obj = hosts.HostManager(nova)
+
+            # Get list of dell nfv nodes
+            dell_hosts = []
+
+            for host in host_obj.list():
+                if "dell-compute" in host.host_name:
```

Comment:
from your approach 
we are getting 
<Server: r175-dell-compute-2>
<Server: r175-dell-compute-1>
<Server: r175-dell-compute-0>

but we wanted 

[u'r175-dell-compute-2.r175.nfv.lab', u'r175-dell-compute-1.r175.nfv.lab', u'r175-dell-compute-0.r175.nfv.lab']

Reply:
ERROR: Property error: : resources.aggregate.properties.hosts[0]: : Error validating value 'r175-dell-compute-0': The Host (r175-dell-compute-0) could not be found.

I think it will requires FQDN.

Reply:
Oh ok i see. There is also another method in nova_client call hypervisors.list() but it only let you get the hypervisor ID not the FQDN of the hosts. I'm still digging in to find something more efficient. However, i think we can go ahead and use your code until we find something else.